c_api.c

```diff
@@ -617,10 +620,11 @@ void free_session_key_list_t(session_key_list_t *session_key_list) {
     free(session_key_list);
 }
 
+void free_session_ctx(SST_session_ctx_t *session_ctx) { free(session_ctx); }
+
```

Comment:
Although there is no recursive frees here, I think we should have this function, or else users will forget (including me...) to free on end.